src/bp/ui-shared/src/ContextMenu/index.tsx

```diff
@@ -0,0 +1,161 @@
+                <Popover
+                    {...this.props}
+                    backdropProps={{ onContextMenu: this.handleBackdropContextMenu }}
+                    content={content}
+                    enforceFocus={false}
+                    key={key}
+                    hasBackdrop={true}
+                    isOpen={this.state.isOpen}
+                    minimal={true}
+                    modifiers={POPPER_MODIFIERS}
+                    onInteraction={this.handlePopoverInteraction}
+                    position={Position.RIGHT_TOP}
+                    popoverClassName={popoverClassName}
+                    target={<div />}
+                    transitionDuration={TRANSITION_DURATION}
+                />
+            </div>
+        );
+    }
+
+    public show(menu: JSX.Element, offset: IOffset, onClose?: () => void, isDarkTheme?: boolean) {
+        this.setState({ isOpen: true, menu, offset, onClose, isDarkTheme });
+    }
+
+    public hide() {
+        safeInvoke(this.state.onClose);
+        this.setState({ isOpen: false, onClose: undefined });
+    }
+
+    private cancelContextMenu = (e: React.SyntheticEvent<HTMLDivElement>) => e.preventDefault();
+
+    private handleBackdropContextMenu = (e: React.MouseEvent<HTMLDivElement>) => {
+        // React function to remove from the event pool, useful when using a event within a callback
+        e.persist();
+        e.preventDefault();
+        // wait for backdrop to disappear so we can find the "real" element at event coordinates.
+        // timeout duration is equivalent to transition duration so we know it's animated out.
+        this.setTimeout(() => {
+            // retrigger context menu event at the element beneath the backdrop.
+            // if it has a `contextmenu` event handler then it'll be invoked.
+            // if it doesn't, no native menu will show (at least on OSX) :(
+            const newTarget = document.elementFromPoint(e.clientX, e.clientY);
+            newTarget.dispatchEvent(new MouseEvent("contextmenu", e));
+        }, TRANSITION_DURATION);
+    };
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => {
+        if (!nextOpenState) {
+            // delay the actual hiding till the event queue clears
+            // to avoid flicker of opening twice
+            requestAnimationFrame(() => this.hide());
+        }
+    };
+}
+
+let contextMenuElement: HTMLElement;
+let contextMenu: ContextMenu;
+*/
+import React, { FC, Fragment, SyntheticEvent, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+import Overlay from '../Overlay'
+
+import style from './style.scss'
+
+const ContextMenuWrapper = ({ event, children }) => {
+  const elPos = event.currentTarget.getBoundingClientRect()
+
+  const handleToggle = e => {
+    e.stopPropagation()
+    removeContextMenu()
+  }
+
+  return (
+    <Fragment>
+      <div
+        style={{ top: `${elPos.top + elPos.height}px`, left: `${elPos.left + elPos.width / 2}px` }}
```

Comment:
Does that mean that context menu will always display in bottom center of the target el ?

Reply:
Yes, at least for now, we can add an overwrite of the positioning if/once we need it in the future. kept it as barebone as possible